plugins/UsersManager/Controller.php

```diff
@@ -405,6 +409,22 @@ public function recordUserSettings()
         return $toReturn;
     }
 
+    public function newsletterSignup()
+    {
+        Piwik::checkUserIsNotAnonymous();
```

Comment:
I think we can add `$this->checkTokenInUrl();` here, though not having it wouldn't be too big of a deal.

Reply:
If I add this the ajax call from the user settings page breaks - I guess I would need to change something in the JS there to force it to pass the token?

Reply:
For `ajaxHelper` I think `withTokenInUrl()` would do the trick.

Reply:
Thanks, working now.